code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -163,3 +163,73 @@
 #undef SPUR_MEDIUM_POWER
 #undef SPUR_LOW_POWER
 #undef SPUR_NO_POWER
+
+/obj/item/weapon/gun/gatling
+	name = "gatling gun"
+	desc = "Ya-ta-ta-ta-ta-ta-ta-ta ya-ta-ta-ta-ta-ta-ta-ta do-de-da-va-da-da-dada! Kaboom-Kaboom!"
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "minigun"
+	item_state = "minigun0"
+	origin_tech = null
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	recoil = 1
+	flags = TWOHANDABLE
+	w_class = 5.0//we be fuckin huge maaan
+	fire_delay = 0
+	var/max_shells = 200
+	var/current_shells = 200
+
+/obj/item/weapon/gun/gatling/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	if(flag)	return //we're placing gun on a table or in backpack
+	if(harm_labeled >= min_harm_label)
+		user << "<span class='warning'>A label sticks the trigger to the trigger guard!</span>" //Such a new feature, the player might not know what's wrong if it doesn't tell them.
+		return
+	if(wielded)
+		Fire(A,user,params, "struggle" = struggle)
+	else
+		user << "<span class='warning'>You must dual-wield \the [src] before you can fire it!</span>"
+
+/obj/item/weapon/gun/gatling/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	..(target,user,params,reflex,struggle)
```

Comment:
This isn't necessary, when you use ..() it automatically takes the arguments here you started with. You would specify if you want to change the arguments